opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -192,24 +207,41 @@ def merge(self, other: "Resource") -> "Resource":
         """
         merged_attributes = self.attributes
         merged_attributes.update(other.attributes)
-        return Resource(merged_attributes)
+
+        if self.schema_url == "":
+            schema_url = other.schema_url
+        elif other.schema_url == "":
+            schema_url = self.schema_url
+        elif self.schema_url == other.schema_url:
+            schema_url = other.schema_url
+        else:
+            schema_url = "merge-conflict"
```

Comment:
This should definitely set the schema URL to an empty string in the event of a conflict.  It should perhaps also return an empty resource since the merged attributes may be semantically invalid. See the [PR in the Go SDK](https://github.com/open-telemetry/opentelemetry-go/pull/1938/files#diff-6b5630aeace530fb9e933030a102bafbf0ceff99d37f7b1c77f7ef1cd7e90ec0R177) from the author of the schema URL spec.